### ISSUE TITLE
cista::vector erase(iterator) on its.end() element stability

### ISSUE THREAD

**Issue by ChemistAion** [felixguendling/cista#142]
I know that cista::vector follows _"Currently, they do not provide exactly the same interface as their std:: equivalents."_ but I would like to report this anyway.

When `cista::vector.erase(iterator)` is updated from its previous result then the vector last element (of more than one in total) will be not _considered_ in such a loop:
```
	CISTA::vector<int> test_cista;
	test_cista.push_back(1);
	test_cista.push_back(2);
	
	for (auto iterator = test_cista.begin(); iterator != test_cista.end(); iterator)
	{
		iterator = test_cista.erase(iterator);
		continue;
	}
```
With one element in `cista::vector` only works fine: https://godbolt.org/z/r8E57hs4s

**Comment by felixguendling** [comment]
Thank you! This should be fixed in https://github.com/felixguendling/cista/commit/3a91caedba6841c0d5141795709affe89015e4de

**Comment by ChemistAion** [comment]
Fix confirmed with MSVC 17.3.1 x64 [Visual Studio 2022 (v143)]
Thank you for prompt action here.
